src/sqlfluff/api/simple.py

```diff
@@ -24,17 +27,22 @@ def _unify_str_or_file(sql):
     return sql
 
 
-def lint(sql, dialect="ansi", rules=None, exclude_rules=None):
+def lint(
+    sql: Union[str, TextIO],
+    dialect: str = "ansi",
+    rules: Optional[Union[str, List[str]]] = None,
+    exclude_rules: Optional[Union[str, List[str]]] = None,
```

Comment:
I used `List` rather than `Iterable` as the linter is already typed to expect a `List`. Simplest to keep as a `List` for now, or we can coerce the supplied `Iterable` to a `List`, or we can follow the rabbit hole down and try and convert the linter type hints to `Iterable` too.

Reply:
Since we're discussing API changes, should we consider removing `Optional` and changing the default value to `""`? `Optional[Union[` illustrates to me that the current API is a bit messy.

Reply:
So actually I personally like the use of None as the default here. My thought is that this should be typed `Optional[List[str])` (or `Optional[Iterable[str]]` in the future) as it's consistent regardless how many rules you provide.

e.g.
```python
results = sqlfluff.lint(sql, rules=["L050"])
```
or
```python
results = sqlfluff.lint(sql, rules=["L050", "L051"])
```
and the optional None is standard for the empty list case.

(I think Optional tends to look bad more as a result of the syntax, once the world moves to 3.10 type hinting it'll look much nicer `list[str] | None` 😄 )

Reply:
^ We should also simplify this downwards into the linter in the future too.